src/wp-includes/user.php

```diff
@@ -1990,7 +2046,103 @@ function validate_username( $username ) {
 	 * @param bool   $valid    Whether given username is valid.
 	 * @param string $username Username to check.
 	 */
-	return apply_filters( 'validate_username', $valid, $username );
+	$valid = apply_filters( 'validate_username', true, $username );
+	if ( ! $valid ) {
+		$errors->add( 'user_name', __( 'Sorry, that username is invalid.' ) );
```

Comment:
We can use the text: "Sorry, that username is not allowed." to maintain the consistency of error messages.